src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaPlayerElementExtensions.wasm.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Windows.UI.Xaml.Controls;
+public class MediaPlayerElementExtensions
+{
+	private static bool _anonymousCORSEnabled;
+	private static MediaPlayerElement _mediaPlayer;
+
+	public static DependencyProperty AnonymousCORSProperty { get; } =
+	DependencyProperty.RegisterAttached(
+		"AnonymousCORS",
+		typeof(bool),
+		typeof(MediaPlayerElementExtensions),
+		new FrameworkPropertyMetadata(false, OnAnonymousCORSChanged));
+
+	public static void SetAnonymousCORS(MediaPlayerElement d, bool value)
+	{
+		d.SetValue(AnonymousCORSProperty, value);
+	}
+
+	private static void OnAnonymousCORSChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+	{
+		if (d is MediaPlayerElement)
+		{
+			_anonymousCORSEnabled = (bool)e.NewValue;
+			_mediaPlayer = (MediaPlayerElement)d;
+			if (_mediaPlayer?.MediaPlayer?.Player != null)
+			{
+				_mediaPlayer.MediaPlayer.Player.SetAnonymousCORS(_anonymousCORSEnabled);
+				return;
+			}
+
+			_mediaPlayer.Loaded -= MediaElement_Loaded;
+			_mediaPlayer.Loaded += MediaElement_Loaded;
+		}
+	}
+
+	private static void MediaElement_Loaded(object sender, RoutedEventArgs e)
+	{
+		_mediaPlayer.MediaPlayer.Player.SetAnonymousCORS(_anonymousCORSEnabled);
+	}
```

Comment:
I'm afraid if multiple media player elements are used and AnonymousCORS property changed for all of them at about the same time.

Then they are all completing on the same state and we might have a race condition.

Reply:
Refactored this part and now it is using a Dictionary to hold the values until related MediaElement object finishes initialization. After that pending CORS value is applied to calling MediaElement and finally it gets removed from pending list.